ecommerce_worker/fulfillment/v1/tasks.py

```diff
@@ -37,25 +38,38 @@ def fulfill_order(self, order_number, site_code=None, email_opt_in=False):
     Returns:
         None
     """
+
+    api_url = urljoin(
+        get_configuration('ECOMMERCE_API_ROOT', site_code=site_code),
+        f'orders/{order_number}/fulfill/'
+    )
     max_fulfillment_retries = get_configuration('MAX_FULFILLMENT_RETRIES', site_code=site_code)
-    api = get_ecommerce_client(site_code=site_code)
+
     try:
         logger.info('Requesting fulfillment of order [%s].', order_number)
-        api.orders(order_number).fulfill.put(email_opt_in=email_opt_in)
-    except exceptions.HttpClientError as exc:
-        status_code = exc.response.status_code  # pylint: disable=no-member
+        access_token = get_access_token()
+
+        headers = {'Authorization': f'JWT {access_token}'}
+        response = requests.put(
+            api_url,
+            params={
+                'email_opt_in': email_opt_in
+            },
+            headers=headers
+        )
+        response.raise_for_status()
+
+    except HTTPError as exc:
+        status_code = exc.response.status_code
         if status_code == 406:
             # The order is not fulfillable. Therefore, it must be complete.
             logger.info('Order [%s] has already been fulfilled. Ignoring.', order_number)
             raise Ignore() from exc
+
         # Unknown client error. Let's retry to resolve it.
         logger.warning(
             'Fulfillment of order [%s] failed because of HttpClientError. Retrying',
```

Comment:
The logging message is not fully correct since we do not work with the HttpClientError any more

Reply:
@ruzniaievdm please exchange `HttpClientError` with `HTTPError`

Reply:
@idegtiarov  `HttpClientError` is a deprecation class from `slumber.exceptions`
in case of use `RequestException` from `requests.exceptions` do not pass tests